lms/djangoapps/instructor/views/api.py

```diff
@@ -726,6 +728,343 @@ def extract_user_info(user):
+        idx += 1
+    return course_tree
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+@require_query_params(rolename="'instructor', 'staff', or 'beta'")
+def list_course_tree(request, course_id):
+    """
+    Returns a tree representing the substructures in a course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_with_access(request.user, 'instructor', course_id, depth=None)
+    course_tree = build_course_tree(course)
+    response_payload = {
+        'course_id': course_id.to_deprecated_string(),
+        'data': course_tree
+    }
+    return JsonResponse(response_payload)
+
+
+def process_new_query(course_id, query_incl, query_type, query_id, query_filtering, entity_name):
+    """
+    Takes a query and bundles it up into a query object
+    """
+    blocks = query_id.split("/")
+    if len(blocks) != 2:
+        return
+    else:
+        block_type, block_id = blocks
+    query_id = course_id.make_usage_key(block_type, block_id)
+    if query_filtering is None:
+        return None
+    query_filtering = query_filtering.lower().strip()
+
+    query_type = query_type.lower().strip()
+    if query_type == "section":
+        query_type = QUERY_TYPE.SECTION
+        if query_filtering == SECTION_FILTERS.OPENED.lower():
+            query_filtering = SECTION_FILTERS.OPENED
+        elif query_filtering == SECTION_FILTERS.NOT_OPENED.lower():
+            query_filtering = SECTION_FILTERS.NOT_OPENED
+        elif query_filtering == SECTION_FILTERS.COMPLETED.lower():
+            query_filtering = SECTION_FILTERS.COMPLETED
+    else:
+        query_type = QUERY_TYPE.PROBLEM
+        if query_filtering == PROBLEM_FILTERS.OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.OPENED
+        elif query_filtering == PROBLEM_FILTERS.NOT_OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_OPENED
+        elif query_filtering == PROBLEM_FILTERS.COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.COMPLETED
+        elif query_filtering == PROBLEM_FILTERS.NOT_COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_COMPLETED
+        elif query_filtering == "score":
+            query_filtering = PROBLEM_FILTERS.SCORE
+        elif query_filtering == "number of peer responses graded":
+            query_filtering = PROBLEM_FILTERS.NUMBER_PEER_GRADED
+    return Query(query_type, query_incl, query_id, query_filtering, entity_name)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_bulk_temp_query(request, course_id):  # pylint: disable=unused-argument
+    """
+    Deletes a temporary query that the user has entered along with the corresponding students
+    """
+    existing = request.GET.get('existing')
+    if existing is not None and existing != "":
```

Comment:
This can be simplified to `if existing:` since both `None` and `""` evaluate to `Falsey`.